lib/ansible/plugins/filter/core.py

```diff
@@ -405,6 +405,9 @@ def comment(text, style='plain', **kw):
         }
     }
 
+    if style not in comment_styles:
+        raise AnsibleFilterError(f"Invalid option '{style}'. Available options: {', '.join(comment_styles.keys())}")
```

Comment:
There's no need to use Ansible-specific exception types in Jinja plugins for simple messages like this, now that DT has been merged. However, in many cases it makes sense to include additional details which can only be provided from an `AnsibleError` derived exception.

In this case, we can use both `obj` and `help_text` to improve the error message users see:

```suggestion
        raise AnsibleTemplatePluginError(
            message=f"Invalid style {style!r}.",
            help_text=f"Available styles: {', '.join(comment_styles)}",
            obj=style,
        )
```

Also, a few other things to keep in mind:

* The Jinja plugin oriented exceptions `AnsibleFilterError` and `AnsibleLookupError` will be deprecated in favor of `AnsibleTemplatePluginError`, so use that instead.
* Be sure to use `!r` to `repr` a string instead of manually quoting it.
* No need to use `.keys()` with `join`.

Reply:
Well, there is one case where even simple messages benefit from using a custom exception type, and that is lookup plugins, which should raise `AnsibleTemplatePluginError` derived errors in situations like this. Otherwise the error will be treated as an "unhandled exception".

That behavior existed before DT, although the exception type was `AnsibleLookupError`, which is now just an alias for `AnsibleTemplatePluginError`.

Reply:
There's an upcoming change that eliminates the special case mentioned above, to further simplify error handling.